### ISSUE TITLE
error: name 'XXX' is not defined

### ISSUE THREAD

**Issue by starbugs-qurong** [exaloop/codon#328]
I have run the following python program program9.py by codon.

class x5Kp54Fa6 (): 
	y8oQ6lg = 25
	RC954Ay7 = "96tcJ4G"
	def u137u306 ( self ) -> int : 
			aOB545rU94count=0
			while (self.RC954Ay7)==None :
				QD31jzVqwcount=QD31jzVqwcount+1
				if QD31jzVqwcount>2:
					break
				Z07144B6 = ( ( self.y8oQ6lg ) * ( 14 ) ) ; 
if __name__ == '__main__':
	OF059yzM= x5Kp54Fa6();
	OF059yzM.u137u306()
	print(OF059yzM.y8oQ6lg)
	print(OF059yzM.RC954Ay7)

And it output error message as follows:
codon_program9.py:7:20-34: error: name 'QD31jzVqwcount' is not defined
codon_program9.py:8:8-22: error: name 'QD31jzVqwcount' is not defined

I hava run it by both python3 and pypy3 and output the following content:
25
96tcJ4G

The related files can be found in https://github.com/starbugs-qurong/python-compiler-test/tree/main/codon/python_9

**Comment by elisbyberi** [comment]
@starbugs-qurong `UnboundLocalError: local variable 'QD31jzVqwcount' referenced before assignment` In Python and Pypy it does not throw an UnboundLocalError exception, because it is dead code. The Python interpreter completely ignores this dead code when executing your functions, but Codon seems to compile it.

@arshajii Should we compile dead code?

**Comment by arshajii** [comment]
Yes this is by design; something like this usually indicates an error in the code and we want Codon to catch it, so we'll keep this behavior for now.
